src/inputs/dnstap/DnstapInputStream.cpp

```diff
@@ -56,15 +60,14 @@ void DnstapInputStream::_read_frame_stream_file()
                 _logger->warn("dnstap data is wrong type or has no message, skipping frame of size {}", len_data);
                 continue;
             }
-            if (_filter_host) {
-                if (d.message().has_query_address()) {
-                    if (!_match_subnet(d.message().query_address())) {
+            if (_f_enabled[Filters::OnlyHosts]) {
+                if (d.message().has_query_address() && d.message().has_response_address()) {
+                    if (!_match_subnet(d.message().query_address()) || !_match_subnet(d.message().response_address())) {
```

Comment:
this `||` should be `&&` because with `||` it means "skip if either doesn't match" but with `&&` it means "skip if both don't match" (i.e. at least one needs to match) and we want the latter. double check that but i'm pretty sure it's right. to be clear: it's ok to have query OR response OR both match, but at least one has to or else we filter.